include/zephyr/devicetree.h

```diff
@@ -17,9 +17,35 @@
 #define DEVICETREE_H
 
 #include <devicetree_generated.h>
-
 #include <zephyr/sys/util.h>
 
+#ifndef _ASMLANGUAGE
+#include <stdint.h>
+#endif
+
+#ifdef __GNUC__
+/*
+ * This macro must produce constant integer expressions, which can't
+ * be done in the preprocessor (casts aren't allowed).  Defer to the
+ * GCC internal functions where they're available.
+ *
+ * native_posix platform uses it's own version of stdint.h, where
+ * this might not be handled, so we explicitly redefine UINT64_C
+ * macro here
+ */
+#undef UINT64_C
+#define UINT64_C(_v) __UINT64_C(_v)
+#endif
```

Comment:
why is this required?

Reply:
Hmmm.. That's very strange, but previously I've got a lot of errors for native_posx platform and associated it with different UINT64_C() macro declaration for that platform.. Now it works flawlessly. So looks like we don't need this hack anymore, thanks for your comment. Removed this